# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/xpconnect/src/nsXPConnect.cpp
# Commit: 33dcf5c2e643
# Full Hash: 33dcf5c2e643b94e90a0a197c99112a080501a12
# Author: Alexandra Borovova <aborovova@mozilla.com>
# Date: 2025-08-20 09:57:21
# Regressor Bug: 1980211
# File Overlap Count: 2
# Description:
#   Bug 1980211 - Attach locale override to Realm instead of setting it in the process. r=dom-core,smaug,tschuster
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D260169
# ==============================================================================

diff -r 2cb7ca176216 -r 33dcf5c2e643 js/xpconnect/src/nsXPConnect.cpp
--- a/js/xpconnect/src/nsXPConnect.cpp	Tue Aug 19 16:57:59 2025 +0000
+++ b/js/xpconnect/src/nsXPConnect.cpp	Tue Aug 19 17:01:50 2025 +0000
@@ -481,7 +481,8 @@
 void InitGlobalObjectOptions(JS::RealmOptions& aOptions,
                              bool aIsSystemPrincipal, bool aSecureContext,
                              bool aForceUTC, bool aAlwaysUseFdlibm,
-                             bool aLocaleEnUS) {
+                             bool aLocaleEnUS,
+                             const nsACString& aLanguageOverride) {
   if (aIsSystemPrincipal) {
     // Make toSource functions [ChromeOnly]
     aOptions.creationOptions().setToSourceEnabled(true);
@@ -501,6 +502,11 @@
     nsCString locale = nsRFPService::GetSpoofedJSLocale();
     aOptions.creationOptions().setLocaleCopyZ(locale.get());
   }
+
+  if (!aLanguageOverride.IsEmpty()) {
+    aOptions.behaviors().setLocaleOverride(
+        PromiseFlatCString(aLanguageOverride).get());
+  }
 }
 
 bool InitGlobalObject(JSContext* aJSContext, JS::Handle<JSObject*> aGlobal,
@@ -555,7 +561,8 @@
   InitGlobalObjectOptions(aOptions, /* aSystemPrincipal */ true,
                           /* aSecureContext */ true,
                           /* aForceUTC */ false, /* aAlwaysUseFdlibm */ false,
-                          /* aLocaleEnUS */ false);
+                          /* aLocaleEnUS */ false,
+                          /* aLanguageOverride */ VoidCString());
 
   // Call into XPCWrappedNative to make a new global object, scope, and global
   // prototype.